support/ci/macos-pkg.sh

```diff
@@ -37,7 +37,7 @@ cat <<- EOF > $src_root/out/hab-bintray.json
   "files": [
     {
       "includePattern": "out/hab-x86_64-darwin.zip",
-      "uploadPattern": "hab-${pkg_version}-${pkg_release}-x86_64-darwin.zip"
+      "uploadPattern": "darwin/x86_64/hab-${pkg_version}-${pkg_release}-x86_64-darwin.zip"
```

Comment:
this just makes the pattern of where the files land match what we do on stable.  Let's us keep the download links and just change s/stable/unstable to get the latest nightly.